app/src/main/java/org/mozilla/focus/session/SessionCallbackProxy.java

```diff
@@ -37,7 +37,10 @@ public void onPageStarted(String url) {
 
     @Override
     public void onPageFinished(boolean isSecure) {
-        session.setLoading(false);
+        if (session.getLoading().getValue() == true) {
```

Comment:
small nit: There's no need to write "== true" here as the result of the method call is already a boolean:
> if (session.getLoading().getValue()) {

Apart from that this is great and seems to work fine. Thank you! :)

Reply:
Oops...